botocore/handlers.py

```diff
@@ -218,8 +218,9 @@ def set_operation_specific_signer(context, signing_name, **kwargs):
         if auth_type == 'v4-unsigned-body':
             context['payload_signing_enabled'] = False
 
-        # s3 and s3-control have customized signers "s3v4" and "s3v4a".
-        if signing_name in ('s3', 's3-outposts'):
+        # Signing names used by s3 and s3-control use customized signers "s3v4"
+        # and "s3v4a".
+        if signing_name in ('s3', 's3-outposts', 's3-object-lambda'):
```

Comment:
Let's hoist this up to something like `S3_SIGNING_NAMES` at the top of the module.

Reply:
Done in 1674a9d